src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -33,6 +34,35 @@ public DataTable GetSchema(string collectionName)
 			return dataTable;
 		}
 
+		public DataTable GetSchema(string collectionName, string[] restrictions)
+		{
+			Action close = null;
+			if (m_connection.State != ConnectionState.Open)
+			{
+				m_connection.Open();
+				close = m_connection.Close;
+			}
```

Comment:
I think this should just rely on the connection already being open, instead of opening and closing the connection (unless this is required for compatibility with other drivers).

Reply:
Ok! Will change it.